source/common/upstream/upstream_impl.cc

```diff
@@ -44,7 +44,8 @@ HostImpl::createConnection(Event::Dispatcher& dispatcher, const ClusterInfo& clu
   Network::ClientConnectionPtr connection =
       cluster.sslContext() ? dispatcher.createSslClientConnection(*cluster.sslContext(), address)
                            : dispatcher.createClientConnection(address);
-  connection->setReadBufferLimit(cluster.perConnectionBufferLimitBytes());
+  uint32_t buffer_limit = cluster.perConnectionBufferLimitBytes();
```

Comment:
nit: const uint32_t, or just lose the local variable and do it inline like it was before (might need static_cast).